oeps/oep-0030-arch-pii-markup-and-auditing.rst

```diff
@@ -0,0 +1,269 @@
+
+.. _Original pull request: https://github.com/edx/open-edx-proposals/pull/
+.. _NIST Special Publication 800-122 (pdf): http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-122.pdf
+
+Abstract
+========
+
+The Open edX platform and related independently deployable applications (IDAs) must store information about learners that identifies the learner and, in some cases, is private to them. As an entity that stores this type of information we strive to exercise caution and respect for the privacy of the learners who use our services. This OEP defines what we consider to be personally identifying information (PII) and the ways in which we keep track of that information. It also introduces processes that will allow us to explicitly document which information stored in our systems is PII and provides mechanisms for helping developers follow the outlined best practices.
+
+This OEP explicitly does not cover:
+
+- The retirement process itself
+- The process for cataloging PII stored in operational logs such as Splunk or New Relic
+- The process for cataloging PII stored in non-operational systems such as analytics databases
+
+Motivation
+==========
+
+The primary goal of this OEP is to allow Open edX administrators and developers to understand which information needs special handling, and identifies our best practices for how to make those designations obvious. Respectful handling of PII depends on the following principles, which must be measured against our ability to run the service and meet our research goals and partner obligations:
+
+#. Storing the minimum personal data necessary
+#. Sharing the minimum personal data necessary
+#. Removing the maximum amount of stored personal information when requested
+#. Ensuring our 3rd party partners remove the maximum amount of stored personal information when requested
+#. Auditing and cataloging the information we already store to ensure that we are meeting our obligations with the data we already store
+#. Guarding against accidental storage of PII
+#. Guarding against adding storage of PII without a means of forgetting it
+
+Definitions
+===========
+
+PII
+---
+Open edX has been using `NIST Special Publication 800-122 (pdf)`_ as our operating guideline for what is and is not PII. In some cases we have decided additional information, such as forum posts, may contain PII and ruled that we should assume the stance that best protects the learner.
+
+Information that is considered PII in Open edX includes, but is not limited to:
+
+- Name (first, last, middle initial, prefix, suffix)
+- Username
+- Password
+- Addresses (mailing, billing, work, etc.)
+- Phone number
+- Email address
+- Date of birth (including partial such as year of birth)
+- IP addresses
+- Job title
+- Gender
+- Sex
+- Social media accounts
+- Website
+- Biography fields (free-form text fields intended to introduce one's self)
+- Forum posts
+- Notes service entries
```

Comment:
Does the inclusion of the last 2 (`Forum posts` and `Notes service entries`) imply that we consider any open-ended text entries as possible PII?  For example, are ORA submissions also in a similar category?

Reply:
Free form text is a bit of a gray area. Where we can we retire it, but ORA submissions have wider ranging implications inside a course and for partner data sharing. @stroilova may have more thoughts on this.